server/sonar-main/src/main/java/org/sonar/application/process/ProcessLauncherImpl.java

```diff
@@ -88,6 +90,22 @@ public ProcessMonitor launch(EsCommand esCommand) {
     }
   }
 
+  private static void cleanupOutdatedEsData(EsCommand esCommand) {
+    EsFileSystem esFileSystem = esCommand.getFileSystem();
+    esFileSystem.getOutdatedDataDirectories().forEach(outdatedDir -> {
+      if (outdatedDir.exists()) {
+        LOG.info("Deleting outdated search index data directory {}", outdatedDir.getAbsolutePath());
+        try {
+          FileUtils.deleteDirectory(outdatedDir);
+        } catch (IOException e) {
+          LOG.info("Failed to delete outdated search index data directory " + outdatedDir.getAbsolutePath(), e);
+        }
+      } else {
+        LOG.trace("No outdated search index data directory found at {}", outdatedDir.getAbsolutePath());
```

Comment:
This logging is actually challengeable. Should I drop it?

Reply:
Yes I think that it should be dropped. It's the regular case.